### PR TITLE
utils: Cancel the conversion tasks if the are older then 24 hours

### PR DESCRIPTION
**Pull request by sayanchowdhury**
Signed-off-by: Sayan Chowdhury <sayan.chowdhury2012@gmail.com>

### CODE REVIEW

**Comment by sayanchowdhury** [comment]
@sinnykumari @dustymabe Please review

File: fedimg/utils.py [hunk 1]
```diff
@@ -205,3 +206,39 @@ def get_image_name_from_ami_name_for_fedmsg(image_name):
...
+        return False
+
+    tasks = output.split('\n')[1:-1]
+
+    for task in tasks:
+        expiration_datetime = datetime.datetime.strptime(
+            task.split()[5],
+            '%Y-%m-%dT%H:%M:%Sz'
+        )
+        import_vol_task_id = task.split[3]
```
Comment (sinnykumari) [inline fedimg/utils.py]: Looks like parenthesis is missing for split() call?

File: fedimg/utils.py [hunk 2]
```diff
@@ -205,3 +206,39 @@ def get_image_name_from_ami_name_for_fedmsg(image_name):
...
+
+        start_datetime = expiration_datetime - datetime.timedelta(days=3)
```
Comment (sinnykumari) [inline fedimg/utils.py]: Any specific reason to pick timedelta value as 3 days?
I am looking at some of the stale task of fedimg and it says expiration date as 2018-07-16 . If we consider timedelta as 3 days, then start_datetime of the task appears to be 2018-07-13 which is not really practical as today's date is 2018-07-11.

**Review by sinnykumari — CHANGES_REQUESTED** [review]

**Comment by sinnykumari** [comment]
Just to confirm, canceling stale conversion task will run when fedimg upload gets triggered for a new compose, correct?

File: fedimg/utils.py (hunk 2 above)
Comment (dustymabe) [reply fedimg/utils.py]: yes it's very odd the API I think reports the time in which the task will be cleaned up maybe ? It is some future time. we need to find out what the time is set to and relate it back to the current time (i.e. create a task and immediately view the date returned from the API (should be something like **X days in the future** or something like that) and then figure out the time the task was created by subtracting X from the date reported from the API.

**Comment by dustymabe** [comment]
> Just to confirm, canceling stale conversion task will run when fedimg upload gets triggered for a new compose, correct?

yes that is what **should** happen.

File: fedimg/utils.py (hunk 2 above)
Comment (sayanchowdhury) [reply fedimg/utils.py]: When I was testing, I tested using a stale data which I had saved. This is a fault on my side, this should have been 7 days as @davdunc had confirmed.

**Comment by dustymabe** [comment]
I'll let @sinnykumari cast my vote in the review :)

**Comment by sinnykumari** [comment]
LGTM

**Pull request merged**